src_c/math.c

```diff
@@ -1287,6 +1293,77 @@ vector_scale_to_length(pgVector *self, PyObject *length)
     Py_RETURN_NONE;
 }
 
+static int
+_vector_move_towards_helper(Py_ssize_t dim, double *origin_coords, double *target_coords, double distance)
+{
+    Py_ssize_t i;
+    double delta[VECTOR_MAX_SIZE];
+    double dist;
+
+    for (i = 0; i < dim; ++i)
+        delta[i] = origin_coords[i] - target_coords[i];
+
+    /* Get magnitude of Vector */
+    dist = sqrt(_scalar_product(delta, delta, dim));
+
+    if (dist == 0)
+        return 1;
+
+    if (dist <= distance)
+    {
+        /* Return target Vector */
+        for (i = 0; i < dim; ++i)
+            origin_coords[i] = target_coords[i];
+        return 1;
+    }
+
+    for (i = 0; i < dim; ++i)
+        origin_coords[i] = origin_coords[i] - delta[i] / dist * distance;
+
+    return 1;
+}
+
+static PyObject *
+vector_move_towards(pgVector *self, PyObject *args)
+{
+    double *target_coords[VECTOR_MAX_SIZE];
+    double distance;
+    pgVector *ret;
+
+    if (!PyArg_ParseTuple(args, "Od:move_towards", &target_coords, &distance))
+        return NULL;
```

Comment:
```c
    PyObject *target;
    pgVector *ret;
    double target_coords[VECTOR_MAX_SIZE];
    double distance;
    pgVector *ret;

    if (!PyArg_ParseTuple(args, "Od:slerp", &target, &distance)) {
        return NULL;
    }

    if (!PySequence_AsVectorCoords(target, target_coords, self->dim)) {
        PyErr_SetString(PyExc_TypeError, "Argument 1 must be a vector.");
        return NULL;
    }
```
The first step is to extract user passed object into a `PyObject *target`. Then, extract contents of that pyobject into C double array `target_coords`, and incase there are errors in this step, a python error is set